usagov/usagov-2021
# Usagov 1745 anchors benefit search

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1745

## Description

Adds fields to the category search page where editors can set anchor IDs for internal linking.
Uses the anchor IDs on the category search pages.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



On a working English benefits search page, edit the page and provide a short identifier for the Life Events ID field and Category Search ID fields. For example:

![image](https://github.com/user-attachments/assets/fbef19ba-b574-4895-8a68-bc0c308ee73d)

Set the body field to have internal links to those anchors:

![image](https://github.com/user-attachments/assets/2bc0389a-c753-4c7e-a488-ca0000bc9f77)

Save the page
### Change Requirements



### Validation Steps

- Given set up above, confirm links to the anchors use the text supplied in the fields.
- Clicking on link to go to the benefit finder tool section scrolls to that section, anchor in URL matches value of the life event grid ID field.
- Clicking on link to go to the search by category section scrolls to that section, anchor in URL matches value of the life category search ID field.
- Confirm validators work, edit the English page. Enter a string that uses spaces and punctuation into each ID field. Confirm an error message is displayed and the field is highlighted when you submit the form.
## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
